api/nodes/views.py

```diff
@@ -187,7 +187,9 @@ def check_branched_from(self, draft):
             raise ValidationError('This draft registration is not created from the given node.')
 
     def check_resource_permissions(self, resource):
-        # Old workflow checks permissions on attached node, not draft
+        # If branched from a node, use the node's contributor permissions. See [ENG-1563]
+        if resource.branched_from_type == 'Node':
```

Comment:
would it be reasonable to raise an exception if `branched_from_type` is somehow something other than `Node` or `DraftNode`?

i'm wary of explicitly handling only one type without checking that the implicit `else` assumption is correct